### PR TITLE
Port test projects to core, use appveyor

### PR DESCRIPTION
**Pull request by TerribleDev**
* Add Appveyor intergration
* Port examples projects to dotnet core
* Port unit tests projects to dotnet core
* Sign assemblies only on release config (it seems linux has yet to get all the bits to do signed builds and so to make this whole repo work on my ubuntu laptop for local dev it has to be done)
* Produce packages versioned from the [APPVEYOR_REPO_TAG environment variable](https://www.appveyor.com/docs/branches/#build-on-tags-github-gitlab-and-bitbucket-only) which is generated by appveyor on tag builds
* Only generate package artifacts on appveyor tag, or if a forcePackage bool has been set.

## SumoLogic.Logging.CI.ps1

This is a small powershell script to run the build, run unit tests (automatically if outside of sumologic) and package nuget packages **if you are in appveyor and the current build is a tag build**. By default this script should be called with no arguments.

You can force artifacts to be created locally by calling the script with a `-Version` flag by itself it will create dll's with the proper version and run the tests if you include the `-forcePackage` flag and set it to 1 you will get nuget packages generated locally. This behavior is off by default as this PR comes with the ability for appveyor, a CI system to build and push packages. However until appveyor is up and running you may need to call the flags yourself.

ex.

`.\SumoLogic.Logging.CI.ps1 -Version 1.1.0 -forcePackage 1` create packages with specified version

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+image: Visual Studio 2017
+version: 1.0.{build}
```
Comment (TerribleDev) [inline appveyor.yml]: this version number is strictly to set the build number

**Comment by latkin** [comment]
Just noticed this one, will review soon

**Comment by latkin** [comment]
Sorry for delay on this one. So what is the nuget update flow now? Does it push a new version on every build, or it's something I opt in to?

**Comment by TerribleDev** [comment]
Every build will do a build and run the unit tests.

To push nuget packages you have to make a  [git tag](https://github.com/TerribleDev/sumologic-net-appenders/blob/porttest/appveyor.yml#L14), with the version number. You can do this either with git, or using the github releases view. You must make the tag's annotation to be the version you wish to release.

For example: `git tag -a 1.1.0 -m "Release some cool new feature"`

OR 

image

image

**Comment by latkin** [comment]
Ok, thanks.

This PR doesn't change the platform targeting of the libraries themselves, just of the test apps?

**Comment by TerribleDev** [comment]
The one and only thing it does do to the actual libraries is change assembly signing to only happen [in release mode](https://github.com/SumoLogic/sumologic-net-appenders/pull/45/files#diff-6af3822b72a00bc5cc46fcfc4ce6a425R19) This is because linux is missing some strong naming bits, so atleast this will let developers work on this repo in *Nix

Outside of that one case its all changes to the examples and unit tests.

**Comment by latkin** [comment]
Awesome, thanks. Now I need to populate the nuget key. How do I do that?

**Comment by TerribleDev** [comment]
https://ci.appveyor.com/tools/encrypt

**Pull request merged**